.gitignore

```diff
@@ -1,2 +1,34 @@
 **/target
 /.idea
+
+# Generated by Cargo
+# will have compiled files and executables
+debug/
+target/
+
+# Remove Cargo.lock from gitignore if creating an executable, leave it for libraries
+# More information here https://doc.rust-lang.org/cargo/guide/cargo-toml-vs-cargo-lock.html
+Cargo.lock
+
+# These are backup files generated by rustfmt
+**/*.rs.bk
+
+# MSVC Windows builds of rustc generate these, which store debugging information
+*.pdb
+
+# MacOS temp file
+.DS_Store
+
+book-test
+guide/book
```

Comment:
These entries are non-necessary because `doc/book` is already covered by `book`.